app/src/main/java/org/kiwix/kiwixmobile/utils/LanguageUtils.kt

```diff
@@ -0,0 +1,231 @@
+  // Factory to the LayoutInflater of the Activity.
+  // The Factory is called on each element name as the xml is parsed and we can therefore modify
+  // the parsed Elements.
+  // A Factory can only be set once on a LayoutInflater. And since we are using the support Library,
+  // which also sets a Factory on the LayoutInflator, we have to access the private field of the
+  // LayoutInflater, that handles this restriction via Java's reflection API
+  // and make it accessible set it to false again.
+  fun changeFont(layoutInflater: LayoutInflater, sharedPreferenceUtil: SharedPreferenceUtil) {
+
+    if (!haveToChangeFont(sharedPreferenceUtil)) {
+      return
+    }
+
+    try {
+      val field = LayoutInflater::class.java.getDeclaredField("mFactorySet")
+      field.isAccessible = true
+      field.setBoolean(layoutInflater, false)
+      layoutInflater.factory = LayoutInflaterFactory(context, layoutInflater)
+    } catch (e: NoSuchFieldException) {
+      Log.w(
+        TAG_KIWIX,
+        "Font Change Failed: Could not access private field of the LayoutInflater",
+        e
+      )
+    } catch (e: IllegalAccessException) {
+      Log.w(
+        TAG_KIWIX,
+        "Font Change Failed: Could not access private field of the LayoutInflater",
+        e
+      )
+    } catch (e: IllegalArgumentException) {
+      Log.w(
+        TAG_KIWIX,
+        "Font Change Failed: Could not access private field of the LayoutInflater",
+        e
+      )
+    }
+  }
+
+  // That's the Factory, that will handle the manipulation of all our TextView's and its subcalsses
+  // while the content is being parsed
+  class LayoutInflaterFactory(
+    private val mContext: Context,
+    private val mLayoutInflater: LayoutInflater
+  ) : LayoutInflater.Factory {
+
+    @SuppressWarnings("ImplicitSamInstance")
+    override fun onCreateView(name: String, context: Context, attrs: AttributeSet): View? {
+
+      // Apply the custom font, if the xml equals "TextView", "EditText" or "AutoCompleteTextView"
+      if (name.equals("TextView", ignoreCase = true) ||
+        name.equals("EditText", ignoreCase = true) ||
+        name.equals("AutoCompleteTextView", ignoreCase = true)
+      ) {
+        try {
+          val view = mLayoutInflater.createView(name, null, attrs)
+          Handler().post {
+            (view as TextView).apply {
+              typeface = Typeface.createFromAsset(
+                mContext.assets,
+                getTypeface(Locale.getDefault().language)
+              )
+              setTextSize(TypedValue.COMPLEX_UNIT_PX, textSize - 2f)
+            }
+          }
+          return view
+        } catch (e: InflateException) {
+          Log.w(TAG_KIWIX, "Could not apply the custom font to $name", e)
+        } catch (e: ClassNotFoundException) {
+          Log.w(TAG_KIWIX, "Could not apply the custom font to $name", e)
+        }
+      }
+      return null
+    }
+  }
+
+  companion object {
+
+    private var mLocaleMap =
```

Comment:
We shouldn't use Hungarian notations. It's there in our guidelines but still they are used in some parts of our code base.

Reply:
yup, I changed a few but overlooked this one, good catch